driver-sync/src/test/functional/com/mongodb/client/Socks5ProseTest.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.client;
+
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.MongoSocketOpenException;
+import com.mongodb.MongoTimeoutException;
+import com.mongodb.connection.ClusterDescription;
+import com.mongodb.connection.ServerDescription;
+import com.mongodb.event.ClusterDescriptionChangedEvent;
+import com.mongodb.event.ClusterListener;
+import org.bson.Document;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.extension.ConditionEvaluationResult;
+import org.junit.jupiter.api.extension.ExecutionCondition;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.lang.String.format;
+import static org.junit.jupiter.api.Assumptions.assumeFalse;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+import static org.mockito.Mockito.atLeast;
+
+/**
+ * See https://github.com/mongodb/specifications/blob/master/source/socks5-support/tests/README.rst#prose-tests
+ */
+@ExtendWith(Socks5ProseTest.SocksProxyPropertyCondition.class)
+class Socks5ProseTest {
```

Comment:
Update.  This may be due to the the fact that the tests run twice in the same Evergreen task. In the Junit output on Evergreen I see 8 passed tests and 12 skipped tests, but in the log file I see 20 passed tests, and 42 skipped tests.  So I'm not sure what's going on.

Reply:
I have divided these tests into separate tasks:

[Auth SOCKS5 proxy: latest Replica Set SSL Linux](https://spruce.mongodb.com/task/mongo_java_driver_socks5_tests__version~latest_os~linux_topology~replicaset_socks_auth~auth_ssl~ssl_test_socks5_patch_80a9cb4f961f727f4a5ac85882e658723e7f04d9_64f28d11a4cf47cbccf3d01d_23_09_02_01_17_06?execution=0&page=0&sortBy=STATUS&sortDir=ASC)
[Auth SOCKS5 proxy: latest Replica Set NoSSL Linux](https://spruce.mongodb.com/task/mongo_java_driver_socks5_tests__version~latest_os~linux_topology~replicaset_socks_auth~auth_ssl~nossl_test_socks5_patch_80a9cb4f961f727f4a5ac85882e658723e7f04d9_64f28d11a4cf47cbccf3d01d_23_09_02_01_17_06?execution=0&sortBy=STATUS&sortDir=ASC)
[NoAuth SOCKS5 proxy: latest Replica Set SSL Linux](https://spruce.mongodb.com/task/mongo_java_driver_socks5_tests__version~latest_os~linux_topology~replicaset_socks_auth~noauth_ssl~ssl_test_socks5_patch_80a9cb4f961f727f4a5ac85882e658723e7f04d9_64f28d11a4cf47cbccf3d01d_23_09_02_01_17_06?execution=0&page=0&sortBy=STATUS&sortDir=ASC)
[NoAuth SOCKS5 proxy: latest Replica Set SSL Linux](https://spruce.mongodb.com/task/mongo_java_driver_socks5_tests__version~latest_os~linux_topology~replicaset_socks_auth~noauth_ssl~nossl_test_socks5_patch_80a9cb4f961f727f4a5ac85882e658723e7f04d9_64f28d11a4cf47cbccf3d01d_23_09_02_01_17_06/tests?execution=0&page=0&sortBy=STATUS&sortDir=ASC)

You can now view all tests. Some tasks may skip tests based on the AUTH_ENABLED setting.

Reply:
Further improved to enhance clarity by hiding 'skipped' tests:

[Auth SOCKS5 proxy: latest Replica Set SSL Linux](https://spruce.mongodb.com/task/mongo_java_driver_socks5_tests__version~latest_os~linux_topology~replicaset_socks_auth~auth_ssl~ssl_test_socks5_patch_80a9cb4f961f727f4a5ac85882e658723e7f04d9_64f7908e56234374344da347_23_09_05_20_33_20?execution=0&page=0&sortBy=STATUS&sortDir=ASC)
[Auth SOCKS5 proxy: latest Replica Set NoSSL Linux](https://spruce.mongodb.com/task/mongo_java_driver_socks5_tests__version~latest_os~linux_topology~replicaset_socks_auth~auth_ssl~nossl_test_socks5_patch_80a9cb4f961f727f4a5ac85882e658723e7f04d9_64f7908e56234374344da347_23_09_05_20_33_20?execution=0&page=0&sortBy=STATUS&sortDir=ASC)
[NoAuth SOCKS5 proxy: latest Replica Set SSL Linux](https://spruce.mongodb.com/task/mongo_java_driver_socks5_tests__version~latest_os~linux_topology~replicaset_socks_auth~noauth_ssl~ssl_test_socks5_patch_80a9cb4f961f727f4a5ac85882e658723e7f04d9_64f7908e56234374344da347_23_09_05_20_33_20?execution=0&sortBy=STATUS&sortDir=ASC)
[NoAuth SOCKS5 proxy: latest Replica Set SSL Linux](https://spruce.mongodb.com/task/mongo_java_driver_socks5_tests__version~latest_os~linux_topology~replicaset_socks_auth~noauth_ssl~nossl_test_socks5_patch_80a9cb4f961f727f4a5ac85882e658723e7f04d9_64f7908e56234374344da347_23_09_05_20_33_20?execution=0&sortBy=STATUS&sortDir=ASC)